package org.example.zentriotesting.model.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.example.zentriotesting.model.entity.response.AppUserDTO;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.authority.SimpleGrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.Collection;
import java.util.List;
import java.util.UUID;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class AppUser implements UserDetails {
    private UUID userId;
    private String username;
    private String email;
    private String password;
    private Gender gender;
    private String profileImage;
    private Boolean isVerified;
    private Boolean isReset;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
//    private RoleName role;


    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
//
        return  null;
    }


    public AppUserDTO toAppUserDTO(AppUser appUser) {
        if(appUser == null) {
            return null;
        }
        return new AppUserDTO(userId,username,gender,email,profileImage,createdAt,updatedAt);
    }

    @Override
    public String getPassword() {
        return password;
    }

    @Override
    public String getUsername() {
        return email;
    }

    @Override
    public boolean isAccountNonExpired() {
        return UserDetails.super.isAccountNonExpired();
    }

    @Override
    public boolean isAccountNonLocked() {
        return UserDetails.super.isAccountNonLocked();
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return UserDetails.super.isCredentialsNonExpired();
    }

    @Override
    public boolean isEnabled() {
        return UserDetails.super.isEnabled();
    }

}

